src/test.c

```diff
@@ -0,0 +1,192 @@
+#include "common.h"
+#include "parseTreeHelpers.h"
+
+// Test functions!
+struct Table* createExampleTable(int count);
+bool destroyExampleTable(struct Table*);
+void testStore(void);
+void testParseGrammer(void);
+void testCreateTable(void);
+void testRetrieve(void);
+
+// The following method stubs are not implemented!
+// When they are implemented, you can remove them
+// from here and link to the full function using our
+// common header file!
+struct ParseTree* parseGrammer(char* sql) {
+  return NULL;
+}
+
+bool storeTuples(char* table, struct Tuple* tuples, int count) {
+  return false;
+}
+
+bool storeTable(struct Table* table) {
+  return false;
+}
+
+/*
+ * This is just the main method!
+ */
+int main(void) {
+  printf("===============Create Table:\n");
+  testCreateTable();
+  printf("===============Example Table:\n");
+
+  printf("\n===============testStore\n");
+  testStore();
+
+  printf("\n===============testRetrieve\n");
+  testRetrieve();
+
+  printf("\n===============testParseGrammer\n");
+  testParseGrammer();
+  return 0;
+}
+
+void testRetrieve(void) {
+  // This depends on testStore being run. because reasons!
+
+  char names[FIELD_SIZE][NAME_LIMIT] = { "name4", "name2", "name3", "name1", };
+  FieldType types[FIELD_SIZE][1] = { INTEGER, TEXT, DATE, INTEGER, };
+  struct Field* projection = createFieldList(names, NULL, types, 4);
+  struct ParseTree* selectCmd = createSelectParseTree("table", projection, NULL);
+
+  struct Table* results = retrieve(selectCmd);
+
+  assert(results->count == 2, "Did not retrieve correct number of records");
```

Comment:
All these test cases, pass! Therefore, we have simple storage and simple retrieval working!